internal/controller/validate.go

```diff
@@ -0,0 +1,46 @@
+// SPDX-License-Identifier:Apache-2.0
+
+package controller
+
+import (
+	v1beta1 "github.com/metallb/frrk8s/api/v1beta1"
+	corev1 "k8s.io/api/core/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// TransientError is an error that happens due to interdependencies
+// between crds, such as referencing non-existing secrets.
+// Since we don't want webhooks to make assumptions on ordering, we reset the
+// fields that could cause a transient error from configurations before validating them.
+type TransientError struct {
+	Message string
+}
+
+func (e TransientError) Error() string { return e.Message }
+
+func Validate(resources ...client.ObjectList) error {
+	clusterResources := ClusterResources{
+		FRRConfigs: make([]v1beta1.FRRConfiguration, 0),
+	}
+
+	for _, list := range resources {
+		if l, ok := list.(*v1beta1.FRRConfigurationList); ok {
+			clusterResources.FRRConfigs = append(clusterResources.FRRConfigs, l.Items...)
+		}
+	}
+	resetTransientFields(clusterResources.FRRConfigs)
```

Comment:
let's call this resetSecrets, we'll change if it grows with multiple fields

Reply:
done